### DISCUSSION TITLE
Flyout at mouse position?

### DISCUSSION THREAD

**Discussion by robkri** [bdlukaa/fluent_ui#621] [Q&A] [answered]
Is it possible to position a Flyout-widget at the current mouse position?

I need a context menu that pops up on right-click (secondary tap) at the position of the curser. I thought, Flyout would be a good choice for that, but as far as I understand, its positioning is very limited. But maybe there's some way I didn't figure out yet?

**Comment by bdlukaa** [comment, marked as answer]
The `Flyout` widget itself is just a helper widget that displays overlays according to the child position. If you want to have an optimized control over the overlay content and position, use the Flutter built-in `Overlay` widget + a `Positioned` widget.

To have more control over the widget position, you can create a custom `RenderBox` as well

**Reply by robkri** [reply]
I will try that. Thank you.

**Comment by klondikedragon** [comment]
@bdlukaa - I've been facing this question as well recently. It does seem like a common need for a desktop app to display a context menu when right clicking at the position of the right click. It seems the Flyout code has been re-worked since the original answer to this was posted, and now there is the optional `position` argument to the `showFlyout` method that looks like gives precise control over the position of the flyout widget.

To capture the current mouse position, I'm using a `Listener` widget that captures mouse pointer position change events as the immediate child widget of the `FlyoutTarget`. Then, in a child widget deep in the widget tree, in an `onLongPress` gesture detector, I'm calling showFlyout and passing as the `position` either the global or local coordinates of the mouse as received by the Listener widget.

Using global mouse position mostly works, except that the flyout appears about 50 pixels too far down and to the right from the mouse cursor. That might be about the height/width of the fluent_ui navbar that the app is using (always shown, in compact mode). I tried making the navbar wider and it widened the gap, so it does make me think that the position I need is relative to the root page (right underneath the navbar), rather than where the FlyoutTarget is, any idea why that might be?

I tried local mouse position (local to the Listener widget that is the immediate child of the FlyoutTarget widget), but that didn't get anywhere close to accurate results.

Any idea how to get the mouse position relative to what showFlyout would expect for its position argument so that it could line up exactly?

**Reply by bdlukaa** [reply]
Are you using more than one navigator (with Navigator 2.0)? If so, you need to calculate the global position according to the navigator. See https://github.com/bdlukaa/fluent_ui/blob/master/example/lib/screens/popups/flyout.dart#L332-L339
